beacon_node/network/src/sync/simple_sync.rs

```diff
@@ -499,14 +503,26 @@ impl<T: BeaconChainTypes> SimpleSync<T> {
             "count" => res.block_bodies.len(),
         );
 
-        self.import_queue
-            .enqueue_bodies(res.block_bodies, peer_id.clone());
+        if !res.block_bodies.is_empty() {
+            // Import all blocks to queue
+            let last_root = self
+                .import_queue
+                .enqueue_bodies(res.block_bodies, peer_id.clone());
+
+            // Attempt to process all recieved bodies by recursively processing the latest block
+            if let Some(root) = last_root {
+                match self.attempt_process_partial_block(peer_id, root, network, &"gossip") {
```

Comment:
Yep I agree, I wasn't sure if `gossip` sent blocks or block roots but since it sends blocks this function won't be called by gossip.
So maybe `sync` or `request` is more appropriate.